starlite/dto/interface.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from abc import ABCMeta, abstractmethod
+from typing import TYPE_CHECKING, Any, Generic, TypeVar
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from starlite.connection import Request
+    from starlite.handlers import BaseRouteHandler
+    from starlite.types import StarliteEncodableType
+    from starlite.types.parsed_signature import ParsedType
+
+__all__ = ("AbstractDTOInterface", "DataT")
+
+DataT = TypeVar("DataT")
```

Comment:
I prefer using `T` and single char for generics, unless there are a lot of them, which there arent.